sdk/servicebus/azure-servicebus/azure/servicebus/_common/_connection_string_parser.py

```diff
@@ -0,0 +1,84 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.servicebus.management._models import DictMixin
+
+class ServiceBusConnectionStringProperties(DictMixin):
+    """
+    Properties of a connection string.
+    """
+    def __init__(self, **kwargs):
+        self._fully_qualified_namespace = kwargs.pop('fully_qualified_namespace', None)
+        self._endpoint = kwargs.pop('endpoint', None)
+        self._entity_path = kwargs.pop('entity_path', None)
+        self._shared_access_signature = kwargs.pop('shared_access_signature', None)
+        self._shared_access_key_name = kwargs.pop('shared_access_key_name', None)
+        self._shared_access_key = kwargs.pop('shared_access_key', None)
+
+    @property
+    def fully_qualified_namespace(self):
+        """The fully qualified host name for the Service Bus namespace.
+        The namespace format is: `<yournamespace>.servicebus.windows.net`.
+        """
+        return self._fully_qualified_namespace
+
+    @property
+    def endpoint(self):
+        """The endpoint for the Service Bus resource. In the format sb://<FQDN>/
+        """
+        return self._endpoint
+
+    @property
+    def entity_path(self):
+        """Optional. Represents the name of the queue/topic.
+        """
+        return self._entity_path
+
+    @property
+    def shared_access_signature(self):
+        """
+        This can be provided instead of the shared_access_key_name and the shared_access_key.
+        """
+        return self._shared_access_signature
+
+
+def parse_connection_string(conn_str):
+    # type(str) -> ServiceBusConnectionStringProperties
+    """Parse the connection string.
+
+    :param conn_str: The connection string that has to be parsed.
+    :type conn_str: str
+    """
+    conn_settings = [s.split("=", 1) for s in conn_str.split(";")]
+    if any(len(tup) != 2 for tup in conn_settings):
+        raise ValueError("Connection string is either blank or malformed.")
+    conn_settings = dict(conn_settings)
+    shared_access_signature = None
+    for key, value in conn_settings.items():
+        if key.lower() == 'sharedaccesssignature':
+            shared_access_signature = value
+    shared_access_key = conn_settings.get('SharedAccessKey')
```

Comment:
I see JS is doing check on shared_access_key and shared_access_key_name should both present:
https://github.com/Azure/azure-sdk-for-js/blob/112bb1da8c66ba90655413e3a76c0a899f2d1dee/sdk/servicebus/service-bus/src/util/connectionStringUtils.ts#L73-L81

we decided not to do this for python?

Reply:
we are still doing this ( on line 69 ). do you want me to remove it?